public/app/plugins/datasource/mysql/sqlCompletionProvider.ts

```diff
@@ -1,22 +1,143 @@
 import {
+  CompletionItemKind,
+  CompletionItemPriority,
   getStandardSQLCompletionProvider,
   LanguageCompletionProvider,
+  LinkedToken,
+  PositionContext,
+  StatementPlacementProvider,
+  SuggestionKind,
+  SuggestionKindProvider,
   TableDefinition,
   TableIdentifier,
+  TokenType,
 } from '@grafana/experimental';
 
 interface CompletionProviderGetterArgs {
-  getMeta: React.MutableRefObject<(t?: TableIdentifier) => Promise<TableDefinition[]>>;
+  getMeta: (t?: TableIdentifier) => Promise<TableDefinition[]>;
 }
 
 export const getSqlCompletionProvider: (args: CompletionProviderGetterArgs) => LanguageCompletionProvider =
   ({ getMeta }) =>
   (monaco, language) => ({
     ...(language && getStandardSQLCompletionProvider(monaco, language)),
-    tables: {
-      resolve: getMeta.current,
-    },
-    columns: {
-      resolve: getMeta.current,
-    },
+    customStatementPlacement,
+    customSuggestionKinds: customSuggestionKinds(getMeta),
   });
+
+export enum CustomStatementPlacement {
```

Comment:
More of a nit: Is there any reason we don't just use constants here?

Reply:
not really.